src/Symfony/Component/Console/Helper/FormatterHelper.php

```diff
@@ -73,6 +73,26 @@ public function formatBlock($messages, $style, $large = false)
     }
 
     /**
+     * Truncates a message to the given length.
+     *
+     * @param string $message
+     * @param int    $length
+     * @param string $suffix
+     *
+     * @return string
+     */
+    public function truncate($message, $length, $suffix = '...')
+    {
+        $computedLength = $length - $this->strlen($suffix);
+
+        if ($computedLength > $this->strlen($message)) {
+            return $message;
+        }
+
+        return mb_substr($message, 0, $length).$suffix;
```

Comment:
the $charset attribute is missing, which makes this function reliant on a global config option. Other places in this component use `mb_detect_encoding($string, null, true)`.